src/api/BaseServiceApi.ts

```diff
@@ -0,0 +1,54 @@
+import { FetchMethods, IBaseServiceApi, ServiceResponseType } from '@/types/serviceTypes'
+import { BaseErrorInstance } from './BaseError'
+import { Data } from '@/types/dataTypes'
+
+export class BaseServiceApi implements IBaseServiceApi {
```

Comment:
export class ApiError extends Error {
   code: string
   description: sring
  constructor(message: string, code, description) {
      super(message)
     this.code = code 
     . ...
  }  
}


BaseServiceApi {
    baseUrl: string
    error: typeof ApiError
     constructor(baseUrl, error = ApiError) {
         this.baseUrl = baseUrl
         this.error= error
     }

   handleFetch() {
     try {res = await}
    catch(e) {
        throw new this.error('message', res.status, e. )
     }
    }
}